artifactory/services/upload.go

```diff
@@ -90,7 +90,13 @@ func (us *UploadService) performUploadTasks(consumer parallel.Runner, uploadSumm
 	if totalFailed > 0 {
 		log.Error("Failed uploading", strconv.Itoa(totalFailed), "artifacts.")
 	}
-	artifactsFileInfo = utils.FlattenFileInfoArray(uploadSummary.FileInfo)
+	if us.ResultWriter != nil {
+		err = us.ResultWriter.Close()
```

Comment:
It makes more sense not to close the ```ResultWriter``` here. The function that creates it (```UploadFilesWithResultReader```) should be the one that closes with defer.
The same should be done in the download flow, in the ```performTasks``` method, in download.go.
This approach is not always possible, but I think it is good practice to stick to this approach when possible (if you opened it? You'll close it :)).